File: apps/web/src/types/analytics.ts
export type AnalysisParameterInfo = {
  name: string;
  required: boolean;
  description: string;
};

export type AnalysisInfo = {
  analysis_name: string;
  short_description: string;
  parameters: AnalysisParameterInfo[];
};

export type ListAnalysesResponse = {
  analyses: AnalysisInfo[];
};

export type RunAnalysisRequest = {
  analysis_name: string;
  parameters?: Record<string, unknown>;
  max_rows?: number;
};

export type RunAnalysisResponse = {
  analysis_name: string;
  sql: string;
  columns: string[];
  rows: unknown[][];
  metadata: {
    row_count: number;
    execution_time_ms: number;
  };
};

File: apps/api/app/api/schemas/analytics.py
from __future__ import annotations

from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field


class AnalysisParameterInfo(BaseModel):
    name: str
    required: bool = False
    description: str


class AnalysisInfo(BaseModel):
    analysis_name: str
    short_description: str
    parameters: List[AnalysisParameterInfo] = []


class ListAnalysesResponse(BaseModel):
    analyses: List[AnalysisInfo]


class RunAnalysisRequest(BaseModel):
    analysis_name: str
    parameters: Dict[str, Any] = Field(default_factory=dict)
    max_rows: Optional[int] = None


class RunAnalysisResponse(BaseModel):
    analysis_name: str
    sql: str
    columns: List[str]
    rows: List[List[Any]]
    metadata: Dict[str, Any]

